polylogue/db.py

```
@@ -122,17 +128,47 @@ def _apply_schema(conn: sqlite3.Connection) -> None:
+        state["conn"] = conn
+        state["path"] = target_path
+        created_here = True
+    state["depth"] += 1
     try:
-        yield conn
+        yield state["conn"]
     finally:
-        conn.close()
+        state["depth"] -= 1
+        if state["depth"] <= 0:
+            if created_here:
```

Ruff (0.15.9)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)